backend/extension/extension_network.go

```diff
@@ -61,11 +61,12 @@ func (n *network) Run(ctx context.Context) {
 
 	for {
 		select {
-		case evtBatch := <-evts:
+		case evtBatch, ok := <-evts:
+			if !ok {
+				log.Infof("evts chan closed")
+				return
+			}
 			n.handleSubnetEvents(evtBatch)
-
-		case <-ctx.Done():
```

Comment:
Why remove this?

Reply:
If we keep this, there is a chance this `for` loop ends before `subnet.WatchLeases` goroutine terminates, then no one will read events from evts chan, `subnet.WatchLeases` goroutine may happen to be writting to it. And it results in `subnet.WatchLeases` hangs forever on `receiver <- batch`.